controllers/argocd/utils.go

```diff
@@ -0,0 +1,46 @@
+// Copyright 2022 KubeSphere Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+package argocd
+
+import (
+	"encoding/json"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+)
+
+// StructToMap converts struct value to map value, which key type is string and value type is interface{}.
+// The type of value must be struct. Any other types will lead an error.
+func StructToMap(value interface{}) (map[string]interface{}, error) {
```

Comment:
`StructToMap` or `InterfaceToMap`?

I didn't see any `struct` in this function.